src/CommonLib/environmentvariablehash.h

```diff
@@ -149,6 +149,95 @@ class ENVIRONMENT_VAR_HASH : public HASH_TABLE<ENVIRONMENT_VAR_ENTRY, PWSTR>
         }
     }
 
+    static
+    VOID
+    SetEnvironmentVariables
+    (
+        ENVIRONMENT_VAR_ENTRY *   pEntry,
+        PVOID                     pvData
+    )
+    {
+        UNREFERENCED_PARAMETER(pvData);
+
+        HRESULT hr = S_OK;
+        DWORD   dwResult = 0;
+        DWORD   dwError = 0;
+        STRU    struNameBuffer;
+        STRU    struValueBuffer;
+        BOOL    fFound = FALSE;
+
+        // pEntry->QueryName includes the trailing =, remove it before calling stru
+        if (FAILED(hr = struNameBuffer.Copy(pEntry->QueryName())))
+        {
+            goto Finished;
+        }
+        dwResult = struNameBuffer.LastIndexOf(L'=');
+        if (dwResult != -1)
+        {
+            struNameBuffer.QueryStr()[dwResult] = L'\0';
+            if (FAILED(hr = struNameBuffer.SyncWithBuffer()))
+            {
+                goto Finished;
+            }
+        }
+
+        if (FAILED(hr = struValueBuffer.Resize(300)))
```

Comment:
you can use STACK_STRU(struValueBuffer, 300) to avoid this Resize call

Reply:
Are you allowed to resize a STACK_STRU after initialization?